subsys/bluetooth/services/bas/bas_bls.c

```diff
@@ -232,6 +246,20 @@ void bt_bas_bls_set_service_required(enum bt_bas_bls_service_required value)
 	bls.additional_status &= ~SERVICE_REQUIRED_MASK;
 	bls.additional_status |= (value << SERVICE_REQUIRED_SHIFT) & SERVICE_REQUIRED_MASK;
 	bt_bas_bls_update_battery_level_status();
+
+	if (IS_ENABLED(CONFIG_BT_BAS_BCS)) {
+		/*
+		 * Set the BCS Immediate Service Required bit as per BAS 1.1 specification
+		 * section 3.4.1.1: The BCS Immediate Service Required bit should be set to true if
+		 * the service Required bit is set to true in the Additional Status field.
+		 */
+		if (value == BT_BAS_BLS_SERVICE_REQUIRED_TRUE) {
+			bt_bas_bcs_set_immediate_service_required(true);
+			return;
+		}
+
+		bt_bas_bcs_set_immediate_service_required(false);
```

Comment:
```suggestion
		if (value == BT_BAS_BLS_SERVICE_REQUIRED_TRUE) {
			bt_bas_bcs_set_immediate_service_required(true);
			return;
		} else if (value != BT_BAS_BLS_SERVICE_REQUIRED_UNKNOWN) {
			bt_bas_bcs_set_immediate_service_required(false);
		}
```

If the service required is unknown it should not trigger a service required change (unless otherwise specified by the service spec)